### PR TITLE
Templates: Added static Wellbeing template

### LINKED ISSUE #1478: Templates: Wellbeing
Create the Wellbeing template from
https://www.figma.com/file/RTzSw47oi6pEZXEfP2JoBo/Templates

### PR DESCRIPTION
**Pull request by mariano-formidable**
**Updated:** 

This PR adds in the Wellbeing template to our app.

This template was particularly difficult to implement because of the limitations of the editor.  In the following ways the template I created is different from figma:

1.)  ~~The design calls for `Source Serif Pro` at a lighter font-weight than the editor supports (for this font).  So, lots of of the titles will appear thicker in the generated template than in the designs.  ~~ 
I'll be able to fix this in a follow up PR when the 300 weight version of the font becomes available.

2.) ~~We cannot upload custom SVGs so instead of using the shapes found in the Wellbeing design, I used the available shapes found in the editor and molded them as best I could to match the design.~~ 

I added the SVGs required for the template to the editors shape library:

(The last 4 are the shapes I added for this template)
image

3.)  Finally, the editor doesn't support transparent shapes with borders so I couldn't create the thin outlines that appear throughout the design.

The thin curve in this image is the effect we can't do with the editor:
image

Fixes #1478 

Relative URL to Wellbeing template:
`/wp-admin/edit.php?post_type=web-story&page=stories-dashboard#/template-detail?id=8&isLocal=false`

image

### CODE REVIEW

**Comment by carloskelly13** [comment]
Could we use just a standard white box for this component?

Screen Shot 2020-05-06 at 7 23 05 PM

**Comment by pbakaus** [comment]
> 1.) The design calls for `Source Serif Pro` at a lighter font-weight than the editor supports (for this font). So, lots of of the titles will appear thicker in the generated template than in the designs.

Why do we not support the lighter font-weight? Is it just not available on Google Fonts, or are we somehow not making it available in the editor?

> 2.) We cannot upload custom SVGs so instead of using the shapes found in the Wellbeing design, I used the available shapes found in the editor and molded them as best I could to match the design.

Why can't we upload custom SVGs? I know we have a problem calculating the size for ones uploaded to the media gallery, but why can't we make it work for templates?

Getting the template shapes into the shape panel is important /cc @samitron7 (but possibly for the v2 PR).

> 3.) Finally, the editor doesn't support transparent shapes with borders so I couldn't create the thing outlines that appear throughout the design.

Do you have an example for this? Is this one of those features that we should add partial support for so that it can display in the editor, but you can't modify?

**Comment by swissspidy** [comment]
Regarding 1) let me know how I can help with the fonts. We should be loading all available variants if possible.

**Comment by swissspidy** [comment]
@mariano-formidable Heads up! I merged the master branch to resolve conflicts. Image URLs are now a bit different, e.g. `${imageBaseUrl}/images/templates/wellbeing/wellbeing_page1_image1.jpg` (dropping `assets/`, and renaming `pluginDir` to `imageBaseUrl`)

**Comment by mariano-formidable** [comment]
> Regarding 1) let me know how I can help with the fonts. We should be loading all available variants if possible.

@swissspidy Here's what I see when I select `Source Serif Pro` in the Editor.  It only gives me 3 options of font-weight, but the figma designs call for a font-weight lighter than "Regular":
image

**Comment by mariano-formidable** [comment]
@pbakaus  Regarding number 2 (about not being able to upload SVGs), when I try and upload an SVG using the media uploader I get this error:
image

I could convert the SVGs in the figma design to images and upload them that way, but then the user's won't be able to change their colors like they can a shape.   But this is definitely v1 of our template design, we've yet to add in animations so we can continue to revamp them until they're ready for release.

**Comment by mariano-formidable** [comment]
@pbakaus  Regarding number 3, in the editor we can only apply a `Fill` to the shape and not a `stroke` so pulling off the effect of a thin wavy line like in the screenshot below with a vector shape (which would be ideal over an image which) isn't currently possible:
image

**Comment by swissspidy** [comment]
> > Why do we not support the lighter font-weight? Is it just not available on Google Fonts, or are we somehow not making it available in the editor?
> 
> It only gives me 3 options of font-weight, but the figma designs call for a font-weight lighter than "Regular":

@pbakaus @mariano-formidable [Source Serif Pro](https://fonts.google.com/specimen/Source+Serif+Pro?query=Source+Serif+Pro) is only available in 400 (regular), 600, and 700 on Google Fonts. Figma uses 300. In the editor we currently only show available font weights. Should we perhaps change that and show all? Not sure how browsers would display that font in 300. Perhaps they won't do anything, which could be weird UX.

**Comment by swissspidy** [comment]
> Why can't we upload custom SVGs? I know we have a problem calculating the size for ones uploaded to the media gallery, but why can't we make it work for templates?

Since we cannot upload SVGs to the media library, we would need to bundle these shapes as part of the plugin, adding them to the currently hardcoded list of masks, no?

**Comment by mariano-formidable** [comment]
> Since we cannot upload SVGs to the media library, we would need to bundle these shapes as part of the plugin, adding them to the currently hardcoded list of masks, no?

I wasn't sure if I should just be adding in new shapes to the shape library willy nilly, but now that I think of it, these shapes are pivotal for this template and they are unique compared to the existing ones (meaning you can just stretch/scale/transform the existing shape to make these new ones)... so okay, I'll figure out how to add them to the hardcoded list and update this template to use them.

**Comment by pbakaus** [comment]
> @pbakaus @mariano-formidable [Source Serif Pro](https://fonts.google.com/specimen/Source+Serif+Pro?query=Source+Serif+Pro) is only available in 400 (regular), 600, and 700 on Google Fonts. Figma uses 300. In the editor we currently only show available font weights. Should we perhaps change that and show all? Not sure how browsers would display that font in 300. Perhaps they won't do anything, which could be weird UX.

That's super strange...I just opened a thread with the fonts team to find out what's going on there.

**Comment by pbakaus** [comment]
> > Since we cannot upload SVGs to the media library, we would need to bundle these shapes as part of the plugin, adding them to the currently hardcoded list of masks, no?
> 
> I wasn't sure if I should just be adding in new shapes to the shape library willy nilly, but now that I think of it, these shapes are pivotal for this template and they are unique compared to the existing ones (meaning you can just stretch/scale/transform the existing shape to make these new ones)... so okay, I'll figure out how to add them to the hardcoded list and update this template to use them.

Yes, perfect. I think the Figma actually outlines all shapes per template that would appear in the shape library, @samitron7 to comment on it.

**Comment by pbakaus** [comment]
re: font: Heard back from Google Fonts team, the relevant blocking bug is https://github.com/google/fonts/issues/23 and it looks like it's getting prioritized (yay!).

**Comment by samitron7** [comment]
1.) Seem this issue is being looked into

2.) and 3) are custom shapes so if we can generate them and bundle these shapes as part of the plugin (as Pascal) suggested that would be great.

**Comment by mariano-formidable** [comment]
> Could we use just a standard white box for this component?
> 
> Screen Shot 2020-05-06 at 7 23 05 PM

@carlos-kelly The circles in the template are setup to be links, but the link popup can't be customized in the editor (yet).  So it looks like this:
wellbeing_links

File: assets/src/edit-story/masks/index.js [hunk 1]
```diff
@@ -60,6 +64,14 @@ const CLIP_PATHS = {
...
+  [MaskTypes.BLOB_6]:
+    'M 0.164 0.780 C 0.124 0.774 0.091 0.738 0.062 0.693 C 0.037 0.655 0.021 0.611 0.015 0.568 C 0.009 0.517 0.010 0.466 0.045 0.439 C 0.100 0.396 0.134 0.363 0.154 0.332 C 0.189 0.276 0.185 0.236 0.195 0.189 C 0.202 0.153 0.215 0.111 0.271 0.060 C 0.301 0.033 0.352 0.019 0.406 0.015 C 0.452 0.012 0.501 0.019 0.548 0.032 C 0.596 0.045 0.640 0.066 0.674 0.095 C 0.711 0.126 0.744 0.164 0.748 0.211 C 0.752 0.250 0.730 0.292 0.703 0.341 C 0.661 0.420 0.653 0.473 0.644 0.515 C 0.632 0.571 0.637 0.615 0.594 0.691 C 0.572 0.732 0.519 0.749 0.465 0.765 C 0.420 0.779 0.369 0.792 0.316 0.793 C 0.263 0.793 0.210 0.787 0.164 0.780 Z',
+  [MaskTypes.BLOB_7]:
+    'M 0.868 0.311 C 0.825 0.118 0.605 0.138 0.509 0.132 C 0.279 0.148 0.217 0.295 0.093 0.595 C -0.030 0.895 0.363 0.890 0.466 0.839 C 0.524 0.810 0.695 0.737 0.730 0.710 C 0.766 0.683 0.912 0.503 0.868 0.311 Z',
+  [MaskTypes.BLOB_8]:
+    'M 0.122 0.334 C 0.241 0.263 0.165 0.188 0.193 0.117 C 0.283 -0.162 0.619 0.117 0.744 0.334 C 0.869 0.550 0.674 0.688 0.549 0.764 C 0.425 0.840 0.395 0.784 0.311 0.670 C 0.226 0.556 0.076 0.623 0.024 0.539 C -0.028 0.454 0.004 0.404 0.122 0.334 Z',
```
Comment (mariano-formidable) [inline assets/src/edit-story/masks/index.js:74]: These blobs are the svgs pulled from the template design.

**Review by BrittanyIRL — APPROVED** [review]
Looks great! I think this is a great v1 of the wellbeing template and it's great that it exposed some things that need to be added/changed on the editor.

**Review by carloskelly13 — APPROVED** [review]

**Pull request merged**